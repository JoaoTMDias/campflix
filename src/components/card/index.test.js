import React from "react";
import { render, cleanup } from "@testing-library/react";
import Card from "./index";

afterEach(cleanup);

describe("<Card />", () => {
  it("renders without error", () => {
    const component = render(
      <Card id="example-id" title="Example title">
        <p>Test children</p>
      </Card>
    );

    expect(component).toMatchSnapshot();
  });

  it("renders a title", async () => {
    const { findByTestId } = render(
      <Card id="example-id" title="Example title">
        <p>Test children</p>
      </Card>
    );

    const title = await findByTestId("card-title");

    expect(title.textContent).toBe("Example title");
  });

  it("renders a child node", async () => {
    const { getAllByTestId } = render(
      <Card id="example-id" title="Example title">
        <p data-testid="child-node">Test children</p>
      </Card>
    );

    const childNode = await getAllByTestId("child-node");

    expect(childNode.length).toBe(1);
  });
});
