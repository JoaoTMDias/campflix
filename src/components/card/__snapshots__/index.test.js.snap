// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        aria-describedby="example-id-title"
        class="sc-AykKC bMVsyO card"
        data-testid="card"
        id="example-id"
      >
        <h3
          class="sc-AykKD fmWgRt card__title"
          data-testid="card-title"
          id="example-id-title"
        >
          Example title
        </h3>
        <p>
          Test children
        </p>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      aria-describedby="example-id-title"
      class="sc-AykKC bMVsyO card"
      data-testid="card"
      id="example-id"
    >
      <h3
        class="sc-AykKD fmWgRt card__title"
        data-testid="card-title"
        id="example-id-title"
      >
        Example title
      </h3>
      <p>
        Test children
      </p>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
